tracee/tracee.go

```diff
@@ -812,6 +814,43 @@ func (t *Tracee) shouldProcessEvent(e RawEvent) bool {
 
 func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 	switch ctx.EventID {
+
+	//capture written files
+	case VfsWriteEventID, VfsWritevEventID:
+		if t.config.CaptureWrite {
+			filePath, ok := args[t.EncParamName[ctx.EventID%2]["pathname"]].(string)
+			if !ok {
+				return fmt.Errorf("error parsing vfs_write args")
+			}
+			// path should be absolute, except for e.g memfd_create files
+			if filePath == "" || filePath[0] != '/' {
+				return nil
+			}
+			dev, ok := args[t.EncParamName[ctx.EventID%2]["dev"]].(int32)
+			if !ok {
+				return fmt.Errorf("error parsing vfs_write args")
+			}
+			inode, ok := args[t.EncParamName[ctx.EventID%2]["inode"]].(int32)
+			if !ok {
+				return fmt.Errorf("error parsing vfs_write args")
+			}
+
+			// Write original filepath mapping to file
+			fileName := fmt.Sprintf("write.dev-%d.inode-%d", dev, inode)
+			destinationFilePath := filepath.Join(t.config.OutputPath, "written_files")
+			f, err := os.OpenFile(destinationFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				return fmt.Errorf("error logging written files")
+			}
+			defer f.Close()
+			if _, err := f.WriteString(fmt.Sprintf("%s %s\n", filePath, fileName)); err != nil {
+				return fmt.Errorf("error logging written files")
+			}
+
+			// index written file by original filepath
+			t.writtenFiles[filePath] = fileName
```

Comment:
This indexing is not used anywhere.
We should avoid writing the same file several times to the index.
I suggest mapping from dev+inode (as the key) to filePath (as the value). 
dev+inode uniquely identify a file in the system.
If a file is deleted, then recreated with the same name, a new inode will be created.
We will write two different files (with different inode numbers), but the path will be the same.